modules/manage/pages/kubernetes/k-rolling-restart.adoc

```diff
@@ -0,0 +1,107 @@
+
+When you run Redpanda in Kubernetes with the Redpanda Operator or the Redpanda Helm chart, your Redpanda cluster is managed as a StatefulSet where each broker runs inside its own Pod. As a result, you can perform a rolling restart using a single command with the Kubernetes API.
+
+== Prerequisites
+
+You must have the following:
+
+* xref:deploy:deployment-option/self-hosted/kubernetes/index.adoc[A Redpanda cluster running in Kubernetes].
+
+include::upgrade:partial$rolling-upgrades/restart-impact.adoc[leveloffset=+1]
+
+== Perform a rolling restart
+
+A rolling restart involves running the following procedure on each broker, one at a time. The Redpanda Operator and the Redpanda Helm chart automate this procedure.
+
+. Place a broker into maintenance mode.
+. Wait for maintenance mode to finish.
+. Terminate the Pod so that it is restarted.
+. Take the broker out of maintenance mode.
+
+When a broker is placed into maintenance mode, it reassigns its partition leadership to other brokers for all topics that have a replication factor greater than one (three is the default replication factor for topics). Reassigning partition leadership involves _draining_ leadership from the broker and _transferring_ that leadership to another broker.
+
+[NOTE]
+====
+The Redpanda Helm chart's `preStop` lifecycle hook puts each broker into maintenance mode before the Pod that it's running on is terminated. If maintenance mode does not complete before the `terminationGracePeriod` the container is forcefully terminated using a `SIGKILL` command.
+
+The default `terminationGracePeriod` is 90 seconds, which should be long enough for large clusters.
+You can test different values in a development environment.
+To configure the `terminationGracePeriod`,
+use the xref:reference:redpanda-helm-spec.adoc#statefulsetterminationgraceperiodseconds[`statefulset.terminationGracePeriodSeconds`] setting.
+====
+
+. Check for topics that have a replication factor greater than one.
++
+Partitions that live on only one broker will be offline during the restart. If you have topics with a replication factor of 1, and if you have sufficient disk space, temporarily xref:manage:data-migration.adoc#change-topic-replication-factor[increase the replication factor] to limit outages for these topics during the rolling upgrade.
+
+. Ensure that the cluster is healthy:
++
+```bash
+kubectl exec <pod-name> --namespace <namespace> -c redpanda -- \
+  rpk cluster health
+```
++
+The draining process won't start until the cluster is healthy.
+The amount of time it takes to drain a broker and reassign partition leadership depends on the number of partitions and how healthy the cluster is.
+For healthy clusters, draining leadership should take less than a minute.
+If the cluster is unhealthy, such as when a follower is not in sync with the leader, then draining the broker can take even longer.
```

Comment:
Can we quantify how much longer? Maybe consider revising the wording here -- "...can take even longer." This feels like a weak statement of fact and would be stronger if we could better quantify the effects of an unhealthy cluster.

Reply:
Thanks, I simplified this content as it doesn't provide any meaningful info for the user to act upon.